customer-service/src/main/scala/domain/HttpClient.scala

```diff
@@ -0,0 +1,21 @@
+package domain
+
+import cats.effect.Async
+import io.circe.generic.auto._
+import org.http4s.Uri
+import org.http4s.circe.CirceEntityCodec.circeEntityDecoder
+import org.http4s.client.{Client, JavaNetClientBuilder}
+
+class HttpClient[F[_]: Async](port: Int, host: String)(client: Client[F]) {
```

Comment:
u didnt recognize that it's just another adapter for another port, let me sketch it

- ClientDetailsRepository { def getDetails(id): OptionT[F, CustomerDetails] }
- `ClientDetailsHttpRepository(HttpConfig(port, host), httpClient: Client[F]) extends ClientDetailsRepository`

technical jargon needs to be separated completely from domain